thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/dao/GenericPojoDao.java

```diff
@@ -440,10 +440,11 @@ public <E extends AbstractDTO> List<E> getV2(final DaoFilter daoFilter) {
               Collections.emptyMap(),
               GenericJsonEntity.class,
               connection));
-      final List<E> results = new ArrayList<>();
+      final List<E> results = new ArrayList<>(entities.size());
       if (CollectionUtils.isNotEmpty(entities)) {
+        Class<E> beanClass = (Class<E>) daoFilter.getBeanClass();
```

Comment:
`final` here is actually important, it may help the compiler a lot

btw were you able to measure the cost of the casting ? in this case JIT should be able to figure this out
anyway code looks good.

let's have a working session about benchmarking and profiling if you're interested